src/file.cr

```diff
@@ -838,6 +838,20 @@ class File < IO::FileDescriptor
     system_truncate(size)
   end
 
+  # Flushes all File's page cache to the disk device so that
+  # all changed information can be retrieved even if the system
+  # crashes or is rebooted. The call blocks until the device reports that
+  # the transfer has completed.
+  # As well as flushing the file data, fsync also flushes the metadata
+  # information associated with the file. To reduce disk activity for
```

Comment:
This is mainly mtime, right? Perhaps mention that.

Reply:
atime, mtime yes. from linux docs:

> fdatasync() is similar to fsync(), but does not flush modified metadata unless that metadata is needed in order to allow a subsequent data retrieval to be correctly handled. For example, changes to st_atime or st_mtime (respectively, time of last access and time of last modification; see stat(2)) do not require flushing because they are not necessary for a subsequent data read to be handled correctly. On the other hand, a change to the file size (st_size, as made by say ftruncate(2)), would require a metadata flush.